container/apptabs.go

```diff
@@ -18,670 +23,396 @@ import (
 //
 // Since: 1.4
 func NewAppTabs(items ...*TabItem) *AppTabs {
-	tabs := &AppTabs{BaseWidget: widget.BaseWidget{}, Items: items, current: -1}
-	if len(items) > 0 {
-		// Current is first tab item
-		tabs.current = 0
-	}
+	tabs := &AppTabs{}
 	tabs.BaseWidget.ExtendBaseWidget(tabs)
-
-	if tabs.mismatchedContent() {
-		internal.LogHint("AppTabs items should all have the same type of content (text, icons or both)")
-	}
-
+	tabs.SetItems(items)
 	return tabs
 }
 
-// NewTabItem creates a new item for a tabbed widget - each item specifies the content and a label for its tab.
-//
-// Since: 1.4
-func NewTabItem(text string, content fyne.CanvasObject) *TabItem {
-	return &TabItem{Text: text, Content: content}
-}
-
-// NewTabItemWithIcon creates a new item for a tabbed widget - each item specifies the content and a label with an icon for its tab.
-//
-// Since: 1.4
-func NewTabItemWithIcon(text string, icon fyne.Resource, content fyne.CanvasObject) *TabItem {
-	return &TabItem{Text: text, Icon: icon, Content: content}
-}
-
-// Append adds a new TabItem to the rightmost side of the tab panel
-func (c *AppTabs) Append(item *TabItem) {
-	c.SetItems(append(c.Items, item))
-}
-
 // CreateRenderer is a private method to Fyne which links this widget to its renderer
-func (c *AppTabs) CreateRenderer() fyne.WidgetRenderer {
-	c.BaseWidget.ExtendBaseWidget(c)
-	r := &appTabsRenderer{line: canvas.NewRectangle(theme.ShadowColor()),
-		underline: canvas.NewRectangle(theme.PrimaryColor()), container: c}
-	r.updateTabs()
+func (t *AppTabs) CreateRenderer() fyne.WidgetRenderer {
+	t.BaseWidget.ExtendBaseWidget(t)
+	r := &appTabsRenderer{
+		baseTabsRenderer: baseTabsRenderer{
+			bar:         &fyne.Container{},
+			divider:     canvas.NewRectangle(theme.ShadowColor()),
+			indicator:   canvas.NewRectangle(theme.PrimaryColor()),
+			buttonCache: make(map[*TabItem]*tabButton),
+		},
+		appTabs: t,
+	}
+	// Initially setup the tab bar to only show one tab, all others will be in overflow.
+	// When the widget is laid out, and we know the size, the tab bar will be updated to show as many as can fit (capped to maxAppTab).
+	r.updateTabs(1)
+	r.updateIndicator()
+	r.applyTheme(t)
 	return r
 }
 
+// Append adds a new TabItem to the end of the tab bar.
+func (t *AppTabs) Append(item *TabItem) {
+	t.SetItems(append(t.Items, item))
+}
+
 // CurrentTab returns the currently selected TabItem.
-func (c *AppTabs) CurrentTab() *TabItem {
-	if c.current < 0 || c.current >= len(c.Items) {
+//
+// Deprecated: Use `AppTabs.Selection() *TabItem` instead.
+func (t *AppTabs) CurrentTab() *TabItem {
+	if t.current < 0 || t.current >= len(t.Items) {
 		return nil
 	}
-	return c.Items[c.current]
+	return t.Items[t.current]
 }
 
 // CurrentTabIndex returns the index of the currently selected TabItem.
-func (c *AppTabs) CurrentTabIndex() int {
-	return c.current
+//
+// Deprecated: Use `AppTabs.SelectionIndex() int` instead.
```

Comment:
`SelectedIndex` is the current state-of-play regarding single selections.

Reply:
That is not what #1395 says

Reply:
Scroll all the way to the end and you will see `Selected` is singular and `Selection` is the (proposed) version for multi. Note also that there are 2 proposals. One includes `Selection` and one does not.

Either way the multi-select portion of that has a) not been implemented and b) does not apply to singular selection widgets.
Can this please match `Radio`, `Select` and `FileIcon` in the current iteration so we do not have an inconsistent API?

Reply:
I don't see `func Selected()` at the end of #1395, can you link to the message you are referring to?

> Can this please match Radio, Select and FileIcon in the current iteration so we do not have an inconsistent API?

So not a func, but an exported `Selected` field?

Reply:
Implemented `func Selected()`